File: signal/R_Scripts/Python_Conversion/Cross-axys.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

accel_data = pd.read_csv(r"C:\Users\tanth\pfee-munic\signal\output.csv")

accel_data['Time'] = range(1, len(accel_data) + 1)
signal_x = accel_data['z']

noise_level = 0.5
noisy_signal_x = accel_data['z'] + np.random.normal(loc=0, scale=noise_level, size=len(accel_data))

model = LinearRegression()
model.fit(accel_data[['x', 'y']], noisy_signal_x)
filtered_signal_x = model.predict(accel_data[['x', 'y']])

plt.figure(figsize=(10, 6))
plt.plot(accel_data['Time'], signal_x, color='blue', linewidth=2, label='Signal X Original')
plt.plot(accel_data['Time'], noisy_signal_x, color='green', linewidth=2, label='Signal X bruité')
plt.plot(accel_data['Time'], filtered_signal_x, color='red', linewidth=2, label='Signal X filtré')
plt.xlabel('Time')
plt.ylabel('Signal Value')
plt.title('Signal X bruité et filtré')
plt.legend(loc='lower right')
plt.grid(True)
plt.show()

File: signal/outdated/R_Scripts/Cross-axys.R
rm(list = ls())


accel_data <- read.csv("C:/Users/tanth/Downloads/accelerometer/accelerometer.csv")
accel_data$Time <- seq(from = 1, to = nrow(accel_data))


signal_x <- accel_data$x


set.seed(123) 
noise_level <- 0.1 
noisy_signal_x <- signal_x + rnorm(length(signal_x), mean = 0, sd = noise_level)

plot(noisy_signal_x, type = 'l', col = 'green', xlab = 'Temps', ylab = 'Valeur du signal', main = 'Signal bruité du capteur x')

cor_matrix <- cor(accel_data[, c("x", "y", "z")])
print(cor_matrix)

model <- lm(noisy_signal_x ~ accel_data$y + accel_data$z)

filtered_signal_x <- fitted(model)

accel_data$filtered_x <- filtered_signal_x


plot(accel_data$Time, noisy_signal_x, type = "l", col = "green", lwd = 2, ylim = range(c(noisy_signal_x, filtered_signal_x), na.rm = TRUE), main = "Signal X bruité et filtré")
lines(accel_data$Time, filtered_signal_x, col = "red", lwd = 2)
legend("topright", legend = c("Signal X bruité", "Signal X filtré"), col = c("green", "red"), lwd = 2)

head(accel_data)